### PR TITLE
Add `LocationChangeAction` type

### PR DESCRIPTION
**Pull request by OliverJAsh**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -6,6 +6,11 @@ declare module 'connected-react-router' {
   interface ConnectedRouterProps {
     history: History
   }
+  
+  export interface LocationChangeAction {
```
Comment (supasate) [inline index.d.ts]: Why do we need to export this interface? This action is supposed to be called internally only.

Comment (OliverJAsh) [reply index.d.ts]: This action will be received in user-defined reducers. Exporting this type allows users to annotate the `action` parameter in their reducers, e.g.: `const reducer = (state, action: LocationChangeAction | MyOtherAction) => state`

**Comment by OliverJAsh** [comment]
Hi, any updates on this?

File: index.d.ts (hunk 1 above)
Comment (tom-s) [reply index.d.ts]: I need the event as well for user-defined reducer (a reducer where i store the whole history). Is there a way to do it without a LOCATION_CHANGE event ?

**Comment by supasate** [comment]
Oops. I just forgot to re-check this PR. @OliverJAsh, can you rebase and resolve the conflict? I'll merge then.

**Comment by OliverJAsh** [comment]
@supasate Done

**Comment by supasate** [comment]
Thanks!

**Pull request merged**